django_admin_lightweight_date_hierarchy/admin.py

```diff
@@ -0,0 +1,78 @@
+import re
+import datetime
+
+from django.conf import settings
+from django.contrib import admin
+from django.utils import timezone
+
+
+def get_date_range_for_heirarchy(date_hierarchy, tz):
+    """Generate date range for date heirarchy.
+
+    date_hierarchy <dict>:
+        year (int, required)
+        month (int)
+        day (int)
+    tz <timezone or None>:
+        When none dates remain naive.
+
+    Returns (tuple):
+        from_date (datetime.datetime, aware if tz is set) inclusive
+        to_date (datetime.datetime, aware if tz is set) exclusive
+    """
+    from_date = datetime.datetime(
+        int(date_hierarchy['year']),
+        int(date_hierarchy.get('month', 1)),
+        int(date_hierarchy.get('day', 1)),
+    )
+
+    if tz:
+        from_date = tz.localize(from_date)
+
+    if 'day' in date_hierarchy:
+        to_date = from_date + datetime.timedelta(days=1)
+
+    elif 'month' in date_hierarchy:
+        assert from_date.day == 1
+        to_date = (from_date + datetime.timedelta(days=32)).replace(day=1)
+
+    else:
+        to_date = from_date.replace(year=from_date.year + 1)
+
+    return from_date, to_date
+
+
+class EnhancedDateHeirarchyListFilter(admin.ListFilter):
+    title = ''
+
+    def __init__(self, request, params, model, model_admin):
+        self.date_hierarchy_field = model_admin.date_hierarchy
```

Comment:
In case someone adds `EnhancedDateHeirarchyListFilter` but does not set `model_admin.date_hierarchy`, it will be `None` (or maybe `AttributeError`) right? Can you catch this and show a nice error message? I can imagine this happening when e.g. refactoring.